worker/util.go

```diff
@@ -88,9 +98,12 @@ func detectResources(conf config.Resources) config.Resources {
 		}
 	}
 
-	if conf.RamGb == 0.0 {
-		res.RamGb = float64(vmeminfo.Total) /
-			float64(1024) / float64(1024) / float64(1024)
+	if conf.Resources.RamGb == 0.0 {
+		res.RamGb = float64(vmeminfo.Total) / float64(gb)
+	}
+
+	if conf.Resources.DiskGb == 0.0 {
+		res.DiskGb = float64(diskinfo.Free) / float64(gb)
```

Comment:
Oh, I missed the recover above. Is nil-pointer the only panic we're catching? Because that's better done by normal checks. If there are potentially other panics because of syscall, then `recover()` makes sense.